garrysmod/gamemodes/sandbox/gamemode/cl_hints.lua

```diff
@@ -1,62 +1,80 @@
 
+local cl_showhints = CreateConVar( "cl_showhints", "1",
+	bit.bor( FCVAR_ARCHIVE, FCVAR_USERINFO ), "Show hints called thru SANDBOX:AddHint", 0, 1 )
 
-CreateClientConVar( "cl_showhints", "1", true, false )
+-- A list of hints we've already done so we don't repeat ourselves
+GM.ProcessedHints = {}
 
--- A list of hints we've already done so we don't repeat ourselves`
-local ProcessedHints = {}
+local function ThrowHint( GM, name, length, sound )
 
---
--- Throw's a Hint to the screen
---
-local function ThrowHint( name )
-
-	local show = GetConVarNumber( "cl_showhints" )
-	if ( show == 0 ) then return end
-
-	if ( engine.IsPlayingDemo() ) then return end
+	if ( !cl_showhints:GetBool() ) then return end
 
 	local text = language.GetPhrase( "Hint_" .. name )
-
-	local s, e, group = string.find( text, "%%([^%%]+)%%" )
-	while ( s ) do
-		local key = input.LookupBinding( group )
-		if ( !key ) then key = "<NOT BOUND>" end
-
-		text = string.gsub( text, "%%([^%%]+)%%", "'" .. key:upper() .. "'" )
-		s, e, group = string.find( text, "%%([^%%]+)%%" )
+	local startpos, endpos, bind = text:find( "%%([^%%]+)%%" )
+
+	if ( startpos ~= nil ) then
+		local buffer = {}
+		local bufferlen = 1
+		local nextstartpos = 1
+		
+		repeat

```

Comment:
Why was the `while` loop replaced with a `repeat` loop? This new code looks infinitely less readable, while seemingly providing absolutely no benefit.